### PR TITLE
feat(CollapsibleCard): update disabled state; add renderTitle prop

### PR DESCRIPTION
**Pull request by akdetrick**
closes NDS-586

- Updates disabled state to use `DisabledShim` (per design)
- Adds `renderTitle` render function to allow complete customization within the clickable title area

We got a bit too specific with our content slots when shipping `CollapsibleCard`, and our design team has been experimenting with slightly different content for this component in some features. 

The `renderTitle` render prop allows developers to define whatever they want as the content of the title area. Implementing this as a render prop function allows the consumer to use the `isOpen` state of the card to conditionally render things.

Screenshot 2024-10-21 at 3 43 46 PM

### CODE REVIEW

File: src/CollapsibleCard/index.js [hunk 1]
```diff
@@ -3,17 +3,21 @@ import PropTypes from "prop-types";
 import cc from "classcat";
 import Row from "../Row";
 import IconButton from "../IconButton";
+import DisabledShim from "../DisabledShim";
+
+const noop = () => {};
```
Comment (akdetrick) [inline src/CollapsibleCard/index.js:8]: noop is a named function - sort of a micro optimization for JS garbage collection (`() => {}` functions are anonymous and redefined each pass)

File: src/CollapsibleCard/index.js [hunk 2]
```diff
@@ -175,7 +185,6 @@ const CollapsibleCard = ({
           onTitleContainerClick(isDisabled, "open");
       }}
       aria-expanded="false"
-      aria-disabled={isDisabled ? "true" : "false"}
```
Comment (akdetrick) [inline src/CollapsibleCard/index.js:178]: This attribute is now handled by `DisabledShim`

File: src/CollapsibleCard/index.stories.js [hunk 3]
```diff
@@ -179,6 +179,28 @@ WithError.args = {
   ),
 };
 
+export const WithCustomTitle = Template.bind({});
+WithCustomTitle.args = {
+  renderTitle: (isOpen) => (
+    <div className="padding--all">
+      This card is <strong>{isOpen ? "open" : "closed"}</strong>.
+    </div>
+  ),
```
Comment (akdetrick) [inline src/CollapsibleCard/index.stories.js:188]: This is what gives us unlimited flexibility moving forward. Collapsible cards are used in a lot of our new designs, and we have a new designer joining soon too. Keep this component flexible will keep it alive and useful as we try out different stuff in design

**Review by blooies — APPROVED** [review]
lgtm!! thank you for adding this functionality to the card!

**Pull request merged**